### PR TITLE
[cxx-interop] NFC: Improve documentation comments

### PR DESCRIPTION
**Pull request by egorzhdan**

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: stdlib/public/Cxx/CxxConvertibleToCollection.swift [hunk 1]
```diff
@@ -38,13 +38,13 @@ internal func forEachElement<C: CxxConvertibleToCollection>(
...
-  /// Creates an array containing the elements of a C++ collection.
+  /// Creates an array containing the elements of a C++ container.
   ///
-  /// This initializer copies each element of the C++ collection to a new Swift
+  /// This initializer copies each element of the C++ container to a new Swift
   /// array.
   ///
   /// - Complexity: O(*n*), where *n* is the number of elements in the C++
-  ///   collection.
+  ///   container.
```
Comment (ravikandhadai) [inline stdlib/public/Cxx/CxxConvertibleToCollection.swift:47]: Wondering if we should add a note (for all these APIs) saying "Note that copying each element of the array may not take constant time, if the type of the element is defined in C++".

File: stdlib/public/Cxx/CxxConvertibleToCollection.swift [hunk 2]
```diff
@@ -38,13 +38,13 @@ internal func forEachElement<C: CxxConvertibleToCollection>(
 }
 
 extension Array {
-  /// Creates an array containing the elements of a C++ collection.
+  /// Creates an array containing the elements of a C++ container.
   ///
-  /// This initializer copies each element of the C++ collection to a new Swift
+  /// This initializer copies each element of the C++ container to a new Swift
   /// array.
   ///
```
Comment (ravikandhadai) [inline stdlib/public/Cxx/CxxConvertibleToCollection.swift:45]: This comment "This initializer copies each element of the C++ container to a new Swift array." might be misinterpreted to mean the init is creating a new Swift Array internally.  You could instead say "Initializes the array (or self) by copying every element of the C++ container".

**Comment by SaiJayanth18** [comment]
Hey, I am new to this, but I want to learn and work to fix. Can I get a explanation of what to do?

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: stdlib/public/Cxx/CxxConvertibleToCollection.swift (hunk 1 above)
Comment (egorzhdan) [reply stdlib/public/Cxx/CxxConvertibleToCollection.swift]: That's reasonable, I added a note about that, thanks.

File: stdlib/public/Cxx/CxxConvertibleToCollection.swift (hunk 2 above)
Comment (egorzhdan) [reply stdlib/public/Cxx/CxxConvertibleToCollection.swift:44]: Done!

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Pull request merged**